xla/tsl/platform/default/dso_loader.cc

```diff
@@ -169,37 +169,37 @@ absl::StatusOr<void*> GetRocblasDsoHandle() {
 }
 
 absl::StatusOr<void*> GetMiopenDsoHandle() {
-  return GetDsoHandle("MIOpen", "");
+  return GetDsoHandle("MIOpen", "1");
 }
 
 absl::StatusOr<void*> GetHipfftDsoHandle() {
-  return GetDsoHandle("hipfft", "");
+  return GetDsoHandle("hipfft", "0");
 }
 
 absl::StatusOr<void*> GetRocrandDsoHandle() {
-  return GetDsoHandle("rocrand", "");
+  return GetDsoHandle("rocrand", "1");
 }
 
 absl::StatusOr<void*> GetRocsolverDsoHandle() {
-  return GetDsoHandle("rocsolver", "");
+  return GetDsoHandle("rocsolver", "0");
 }
 
 #if TF_ROCM_VERSION >= 40500
 absl::StatusOr<void*> GetHipsolverDsoHandle() {
-  return GetDsoHandle("hipsolver", "");
+  return GetDsoHandle("hipsolver", "1");
 }
 #endif
 
 absl::StatusOr<void*> GetRoctracerDsoHandle() {
-  return GetDsoHandle("roctracer64", "");
+  return GetDsoHandle("roctracer64", "4");
 }
 
 absl::StatusOr<void*> GetHipsparseDsoHandle() {
-  return GetDsoHandle("hipsparse", "");
+  return GetDsoHandle("hipsparse", "1");
 }
 
 absl::StatusOr<void*> GetHipblasltDsoHandle() {
-  return GetDsoHandle("hipblaslt", "");
+  return GetDsoHandle("hipblaslt", "0");
```

Comment:
hipsolver and hipblaslt need to change based on rocm version, please check https://github.com/ROCm/xla/pull/222

Reply:
When I check inside ROCm libs I see "" and "0" as versions though:
`libhipblaslt.so             libhipblaslt.so.0           libhipblaslt.so.0.15.70000`

Reply:
So that PR does not seem right. When I look into ROCm 7, I see that hipblaslt has version "0". 
Should I cherry-pick that PR as well and then correct the version? I remember you told that it is not needed but we may need it then for the conditional.